gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dweaponpreview_ttt2.lua

```diff
@@ -0,0 +1,377 @@
+
+		cEnt:SetNoDraw(true)
+		cEnt:SetIK(false)
+
+		cEnt:SetParent(self.data.ply)
+		cEnt:AddEffects(EF_BONEMERGE)
+
+		if wep.HoldType ~= "normal" then
+			self.data.wep = cEnt
+		end
+	end
+
+
+	self.data.HoldType = wep.HoldType
+
+	if wep.HoldType == "normal" then
+		self.data.ply:SetSequence(self.data.ply:LookupSequence("idle_all_01"))
+	else
+		self.data.ply:SetSequence(self.data.ply:LookupSequence("idle_" .. wep.HoldType))
+	end
+
+	self.data.ply:SetupBones()
+end
+
+---
+-- @realm client
+function PANEL:DrawModel()
+	local ply = self.data.ply
+	local wep = self.data.wep
+
+	if not IsValid(ply) then return end
+
+	local w, h = self:GetSize()
+	local xBaseStart, yBaseStart = self:LocalToScreen(0, 0)
+
+	local xLimitStart, yLimitStart = xBaseStart, yBaseStart
+	local xLimitEnd, yLimitEnd = self:LocalToScreen(self:GetWide(), self:GetTall())
+
+	local curparent = self
+
+	-- iterate till the top is found to make sure the image is not out of bounds
+	while curparent:GetParent() do
+		curparent = curparent:GetParent()
+
+		local x1, y1 = curparent:LocalToScreen(0, 0)
+		local x2, y2 = curparent:LocalToScreen(curparent:GetWide(), curparent:GetTall())
+
+		xLimitStart = mathMax(xLimitStart, x1)
+		yLimitStart = mathMax(yLimitStart, y1)
+		xLimitEnd = mathMin(xLimitEnd, x2)
+		yLimitEnd = mathMin(yLimitEnd, y2)
+	end
+
+	-- This adds support for the "SWEP Construction Kit" that is used by many weapons such as the
+	-- boomerang or the melon mine: https://steamcommunity.com/sharedfiles/filedetails/?id=109724869
+	--
+	-- This code is taken and slightly modified from its WorldModel renderer
+	if self.data.specialModelHandeling then
+		local wepEnt = weapons.Get(self.data.wepClass)
+		local pos, ang
+
+		for name, worldModelData in pairs(self.data.worldModelData) do
+			wep = self.data.worldModels[name]
+
+			if worldModelData.bone then
+				pos, ang = wepEnt:GetBoneOrientation(wepEnt.WElements, worldModelData, self.data.ply)
+			else
+				pos, ang = wepEnt:GetBoneOrientation(wepEnt.WElements, worldModelData, self.data.ply, "ValveBiped.Bip01_R_Hand")
+			end
+
+			wep:SetPos(pos + ang:Forward() * worldModelData.pos.x + ang:Right() * worldModelData.pos.y + ang:Up() * worldModelData.pos.z)
+
+			ang:RotateAroundAxis(ang:Up(), worldModelData.angle.y)
+			ang:RotateAroundAxis(ang:Right(), worldModelData.angle.p)
+			ang:RotateAroundAxis(ang:Forward(), worldModelData.angle.r)
+
+			wep:SetAngles(ang)
+
+			local matrix = Matrix()
+			matrix:Scale(worldModelData.size)
+			wep:EnableMatrix("RenderMultiply", matrix)
+
+			wep:SetMaterial(worldModelData.material)
+
+			if worldModelData.skin and worldModelData.skin ~= wep:GetSkin() then
+				wep:SetSkin(worldModelData.skin)
+			end
+
+			if worldModelData.bodygroup then
+				for bodygroup, value in pairs(worldModelData.bodygroup) do
+					if wep:GetBodygroup(bodygroup) ~= value then
+						wep:SetBodygroup(bodygroup, value)
+					end
+				end
+			end
+		end
+	end -- end of SWEP Construction Kit support
```

Comment:
replace with the renderer from PR #1325

Reply:
Mhh yeah lets halt it and just use the other function once that is merged, which is making good progress